File: ws/morillo/u2/ws24 Zoo/Zoo/src/main/java/ec/edu/espe/zoo/model/Lion.java
package ec.edu.espe.zoo.model;

import java.util.Date;

/**
 *
 * @author David Morillo, JavaSquad, DCCO-ESPE
 */
public class Lion extends Feline{
    
    private boolean mane;

    public Lion(boolean mane, int id, String description, Date dateOnborn, char gender, int numberOfBones, Cage cage) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
        this.mane = mane;
    }

    @Override
    public String toString() {
        return "Lion{" + "mane= " + mane + super.toString() + '}';
    }
    
    @Override
    public void Feed(int food) {
        System.out.println("Feeding the Lion with " + food + " antelope");
    }
    
    public void feed(int quantity, Food food){
        System.out.println("Feeding the Lion with " + quantity + " " + food);
        System.out.println("Number of legs " + numberOfLegs);
    }

    /**
     * @return the mane
     */
    public boolean isMane() {
        return mane;
    }

    /**
     * @param mane the mane to set
     */
    public void setMane(boolean mane) {
        this.mane = mane;
    }
    
}

File: hw/ccedeno/u2/HW18ZooInPythonJavaScript/ZooInJavaScript/ec.edu.espe.Zoo.model/Lion.js
const Feline = require('../ec.edu.espe.Zoo.model/Feline');
const Food = require('../ec.edu.espe.Zoo.model/Food');

class Lion extends Feline {
  constructor(mane, id, description, dateOnborn, gender, numberOfBones, cage) {
    super(id, description, dateOnborn, gender, numberOfBones, cage);
    this.mane = mane;
  }

  toString() {
    return `Lion{mane=${this.mane}, ${super.toString()}}`;
  }

  feed(food) {
    console.log("Feeding a Lion with " + food + " rats");
  }

  feedWithFood(quantity, food) {
    console.log(`Feeding the lion with ${quantity} ${food}(s)`);
    console.log(`Number of legs ${this.numberOfLegs}`);
  }

  // Getters and Setters
  isMane() {
    return this.mane;
  }

  setMane(mane) {
    this.mane = mane;
  }
}

module.exports = Lion;